### PR TITLE
fix(plugin-meetings): should throw user in lobby

### PR DESCRIPTION
**Pull request by rajeshtezu90**
# COMPLETES #<[SPARK-249834](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-249834) >

## This pull request addresses

Fixes the issue when user is in lobby of a PMR meeting and tries to add media. Earlier it used to skip that check and go for SDP exchange where it was finally failing.

## by making the following changes

Fixes the issue and throws proper error.

Screenshot
in_lobby

In the above screenshot we are already showing Admit button implying the user is in lobby

Unit test

lobby-test

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by Neeraj-swarnkar — APPROVED** [review]
Looks Good.

**Comment by arun3528** [comment]
@rajeshtezu90  can we add some test

**Comment by rajeshtezu90** [comment]
> @rajeshtezu90 can we add some test

@arun3528 Unit test for this case was already there, updated as per the code change.

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 1]
```diff
@@ -4175,7 +4175,7 @@ export default class Meeting extends StatelessWebexPlugin {
       return Promise.reject(new UserNotJoinedError());
     }
     // If the user is unjoined or guest waiting in lobby dont allow the user to addMedia
-    if (this.guest && MeetingUtil.isUserInIdleState(this.locusInfo) && !this.wirelessShare) {
+    if (MeetingUtil.isUserInIdleState(this.locusInfo) && !this.wirelessShare) {
```
Comment (marcin-bazyl) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:4178]: this change implies that locus self.state being IDLE means that user is in a lobby. Has this been confirmed with Locus team or maybe there are some Locus docs that confirm this?

Comment (rajeshtezu90) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:4178]: @arun3528 Could you please confirm this?

Comment (rajeshtezu90) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Added check for self.state being IDLE and intent type WAIT as per the confirmation from locus team.

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/selfUtils.js [hunk 2]
```diff
@@ -56,7 +56,8 @@ SelfUtils.parse = (self, deviceId) => {
       selfIdentity: SelfUtils.getSelfIdentity(self),
       selfUrl: self.url,
       removed: self.removed,
-      roles: SelfUtils.getRoles(self)
+      roles: SelfUtils.getRoles(self),
+      isUserUnadmitted: self.state === _IDLE_ && joinedWith?.intent?.type === _WAIT_,
```
Comment (rajeshtezu90) [inline packages/node_modules/@webex/plugin-meetings/src/locus-info/selfUtils.js:60]: @arun3528,  Reason to not use `SelfUtils.isUserUnadmitted()` is when we pass `self` object from here, it doesn't contain `joinedWith` property which is being used inside the method.

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 3]
```diff
@@ -4175,7 +4175,7 @@ export default class Meeting extends StatelessWebexPlugin {
       return Promise.reject(new UserNotJoinedError());
     }
     // If the user is unjoined or guest waiting in lobby dont allow the user to addMedia
-    if (this.guest && MeetingUtil.isUserInIdleState(this.locusInfo) && !this.wirelessShare) {
+    if (this.isUserUnadmitted && MeetingUtil.isUserInIdleState(this.locusInfo) && !this.wirelessShare) {
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:4178]: we dont need the isUserInIdleState now

**Review by arun3528 — APPROVED** [review]

**Pull request merged**